apps/web/src/components/Publication/OpenAction/UnknownModule/Decent/CurrencySelector.tsx

```diff
@@ -0,0 +1,198 @@
+                wethBalanceData?.decimals as number
+              )
+            ).toFixed(2),
+            usd: (
+              parseFloat(
+                formatUnits(
+                  wethBalanceData?.value as bigint,
+                  wethBalanceData?.decimals as number
+                )
+              ) * wethPriceUsd
+            ).toFixed(2)
+          }
+        : { token: 0, usd: 0 },
+    WMATIC:
+      wmaticBalanceData && wmaticPriceUsd
+        ? {
+            token: parseFloat(
+              formatUnits(
+                wmaticBalanceData?.value as bigint,
+                wmaticBalanceData?.decimals as number
+              )
+            ).toFixed(2),
+            usd: (
+              parseFloat(
+                formatUnits(
+                  wmaticBalanceData?.value as bigint,
+                  wmaticBalanceData?.decimals as number
+                )
+              ) * wmaticPriceUsd
+            ).toFixed(2)
+          }
+        : { token: 0, usd: 0 }
+  };
+
+  const isLoading =
+    wmaticBalanceLoading ||
+    wethBalanceLoading ||
+    usdcBalanceLoading ||
+    wmaticPriceLoading ||
+    wethPriceLoading ||
+    usdcPriceLoading;
+
+  return (
+    <div className="flex h-[80vh] w-full flex-col gap-2 p-5">
+      {allowedTokens
+        .filter((t) => OA_SUPPORTED_CURRENCIES.includes(t.symbol))
+        .map((token) => {
+          return (
+            <div
+              className="flex w-full cursor-pointer items-center justify-between rounded-lg p-2 hover:bg-gray-500/10"
+              key={token.symbol}
+              onClick={(e) => {
+                e.stopPropagation();
+                onSelectCurrency(token.contractAddress as Address);
+              }}
+            >
+              <div className="flex items-center gap-2">
+                <div className="relative">
+                  <img
+                    alt={token.symbol}
+                    className="size-10"
+                    height={40}
+                    src={getTokenImage(token.symbol)}
+                    title={token.name}
+                    width={40}
+                  />
+                  <img
+                    alt="Polygon"
+                    className="absolute bottom-0 right-0"
+                    height={16}
+                    src={`${STATIC_IMAGES_URL}/chains/polygon.svg`}
+                    title="Polygon"
+                    width={16}
+                  />
+                </div>
+                <div className="flex flex-col items-start justify-center leading-none">
+                  <p>{token.symbol}</p>
+                  <p className="text-sm opacity-50">Polygon</p>
+                </div>
+              </div>
+
+              <div className="flex flex-col items-end justify-center gap-1 leading-none">
+                {isLoading ? (
+                  <div className="animate-shimmer h-4 w-16 rounded-lg bg-gray-200" />
+                ) : (
+                  <p>
```

Comment:
animate-shimmer can be replaced with just shimmer, we have a custom css that handles dark mode as well